### PR TITLE
[Docs] How to use cloud storage

### PR DESCRIPTION
**Pull request by eaudetcobello**
This document explains how to use cloud storage with the standalone k8s snap. At the end of the document, the user is able to deploy a pod that uses EBS volumes for storage.

### CODE REVIEW

File: docs/src/snap/howto/storage/cloud.md [hunk 1]
```diff
@@ -0,0 +1,476 @@
...
+        "iam:CreateServiceLinkedRole",
+        "kms:DescribeKey"
+      ],
+      "Resource": [
+        "*"
+      ]
+    }
+  ]
+}
+```
```
Comment (HomayoonAlimohammadi) [inline docs/src/snap/howto/storage/cloud.md:101]: Maybe we can also include the commands to create a role, create these policies, attach these policies to that role and finally attach that role to an instance?

File: docs/src/snap/howto/storage/cloud.md [hunk 2]
```diff
@@ -0,0 +1,476 @@
...
+The cloud controller manager uses the node name to correctly associate the node
+with an EC2 instance. In Canonical K8s, the node name is derived from the
+hostname of the machine. Therefore, before bootstrapping the cluster, we must
+first set an appropriate host name.
+
+```bash
+echo "$(sudo cloud-init query ds.meta_data.local-hostname)" | sudo tee /etc/hostname
+```
+
+Then, reboot the machine.
```
Comment (HomayoonAlimohammadi) [inline docs/src/snap/howto/storage/cloud.md:141]: Maybe we can add the command here?

File: docs/src/snap/howto/storage/cloud.md [hunk 3]
```diff
@@ -0,0 +1,476 @@
...
+look like:
+
+```bash
+ip-172-31-11-86.us-east-2.compute.internal
+```
+
+This host name format is called IP-based naming and is specific to AWS.
+
+```{note} Don't rely on the PS1 prompt to know if your host name was changed
+successfully. The PS1 prompt only displays the hostname up to the first `.`.
```
Comment (HomayoonAlimohammadi) [inline docs/src/snap/howto/storage/cloud.md:153]: Nice! Advanced stuff :D TIL

File: docs/src/snap/howto/storage/cloud.md (hunk 1 above)
Comment (HomayoonAlimohammadi) [reply docs/src/snap/howto/storage/cloud.md:101]: Or maybe not, we can argue that we shouldn't include aws commands in our docs for various reasons. 
Your choice tho, feel free to ignore this.

File: docs/src/snap/howto/storage/cloud.md [hunk 4]
```diff
@@ -0,0 +1,476 @@
...
+```
+
+Finally, install the Helm chart, making sure to set the correct region as an
+argument.
+
+```bash
+sudo k8s helm upgrade --install aws-ebs-csi-driver \
+    --namespace kube-system \
+    aws-ebs-csi-driver/aws-ebs-csi-driver \
+    --set controller.region=us-east-2
```
Comment (HomayoonAlimohammadi) [inline docs/src/snap/howto/storage/cloud.md:403]: Maybe we can put region as a template here like `<REGION>`

File: docs/src/snap/howto/storage/cloud.md [hunk 5]
```diff
@@ -0,0 +1,476 @@
...
+  - name: app
+    image: nginx
+    volumeMounts:
+    - mountPath: "/data"
+      name: ebs-volume
+  volumes:
+  - name: ebs-volume
+    persistentVolumeClaim:
+      claimName: ebs-pvc
+```
```
Comment (HomayoonAlimohammadi) [inline docs/src/snap/howto/storage/cloud.md:472]: Maybe we can also show commands to create the StorageClass, PVC and the Pod.

**Review by HomayoonAlimohammadi — APPROVED** [review]
Great work! Thanks a lot @eaudetcobello! LGTM, just some minor suggestions.

File: docs/src/snap/howto/storage/cloud.md [hunk 6]
```diff
@@ -0,0 +1,476 @@
...
+    }
+  ]
+}
+```
+
+
+## Set your host name
+
+The cloud controller manager uses the node name to correctly associate the node
+with an EC2 instance. In Canonical K8s, the node name is derived from the
```
Comment (evilnick) [inline docs/src/snap/howto/storage/cloud.md:133]: {{product}}

File: docs/src/snap/howto/storage/cloud.md [hunk 7]
```diff
@@ -0,0 +1,476 @@
...
+```
+
+This host name format is called IP-based naming and is specific to AWS.
+
+```{note} Don't rely on the PS1 prompt to know if your host name was changed
+successfully. The PS1 prompt only displays the hostname up to the first `.`.
+```
+
+
+## Bootstrap Canonical K8s
```
Comment (evilnick) [inline docs/src/snap/howto/storage/cloud.md:157]:
```suggestion
## Bootstrap {{product}}
```

File: docs/src/snap/howto/storage/cloud.md [hunk 8]
```diff
@@ -0,0 +1,476 @@
...
+
+Now that your machine has an appropriate host name, you are ready to bootstrap
+Canonical K8s.
```
Comment (evilnick) [inline docs/src/snap/howto/storage/cloud.md:160]: you know what I am going to say...

File: docs/src/snap/howto/storage/cloud.md [hunk 9]
```diff
@@ -0,0 +1,476 @@
...
+Then, bootstrap the cluster:
+
+```bash
+sudo k8s bootstrap --file ./bootstrap-config.yaml
+sudo k8s status --wait-ready
+```
+
+## Deploy the cloud controller manager
+
+Now that you have an appropriate host name, policies, and a Canonical K8s
```
Comment (evilnick) [inline docs/src/snap/howto/storage/cloud.md:179]:
```suggestion
Now that you have an appropriate host name, policies, and a {{product}}
```

File: docs/src/snap/howto/storage/cloud.md [hunk 10]
```diff
@@ -0,0 +1,476 @@
...
+to take advantage of cloud storage solutions in the context of Kubernetes.
+
+## What you'll need
+
+This guide assumes the following:
+
+- You have root or sudo access to an Amazon EC2 instance
+- You can create roles and policies in AWS
+
+
```
Comment (evilnick) [inline docs/src/snap/howto/storage/cloud.md:15]: I suggest we say this is explicitly for AWS. In future we can use tabs to add Azure etc

File: docs/src/snap/howto/storage/cloud.md (hunk 1 above)
Comment (eaudetcobello) [reply docs/src/snap/howto/storage/cloud.md:101]: I would rather leave that up to the user. I use the UI personally.

File: docs/src/snap/howto/storage/cloud.md (hunk 4 above)
Comment (eaudetcobello) [reply docs/src/snap/howto/storage/cloud.md:403]: Sure

File: docs/src/snap/howto/storage/cloud.md (hunk 5 above)
Comment (eaudetcobello) [reply docs/src/snap/howto/storage/cloud.md:472]: Not sure what you mean. Those objects are created by applying the YAML.

File: docs/src/snap/howto/storage/cloud.md [hunk 11]
```diff
@@ -0,0 +1,486 @@
...
+```
+
+## Deploy the cloud controller manager
+
+Now that you have an appropriate host name, policies, and a {{product}}
+cluster, you have everything you need to deploy the cloud controller manager.
+
+Here is a YAML definition file that sets appropriate defaults for you, it
+configures the necessary service accounts, roles, and daemonsets:
+
```
Comment (nhennigan) [inline docs/src/snap/howto/storage/cloud.md:194]: should you include a kubectl apply -f command to deploy the controller manager?

File: docs/src/snap/howto/storage/cloud.md (hunk 5 above)
Comment (nhennigan) [reply docs/src/snap/howto/storage/cloud.md:482]: I think Hue was saying to include the apply yaml command?

**Review by nhennigan — COMMENTED** [review]
Looks great!

File: docs/src/snap/howto/storage/cloud.md (hunk 11 above)
Comment (eaudetcobello) [reply docs/src/snap/howto/storage/cloud.md:194]: Ah, that was an oversight on my part. Thanks.

**Review by nhennigan — APPROVED** [review]

**Pull request merged**